components/playlist/browser/playlist_service.cc

```diff
@@ -0,0 +1,575 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/playlist/browser/playlist_service.h"
+
+#include <algorithm>
+#include <utility>
+
+#include "base/bind.h"
+#include "base/containers/flat_set.h"
+#include "base/files/file_enumerator.h"
+#include "base/files/file_util.h"
+#include "base/json/json_reader.h"
+#include "base/json/json_writer.h"
+#include "base/strings/utf_string_conversions.h"
+#include "base/task/post_task.h"
+#include "base/task_runner_util.h"
+#include "base/token.h"
+#include "brave/components/playlist/browser/playlist_constants.h"
+#include "brave/components/playlist/browser/playlist_data_source.h"
+#include "brave/components/playlist/browser/playlist_service_observer.h"
+#include "brave/components/playlist/common/pref_names.h"
+#include "components/prefs/pref_service.h"
+#include "components/user_prefs/user_prefs.h"
+#include "content/public/browser/browser_context.h"
+#include "content/public/browser/storage_partition.h"
+#include "services/network/public/cpp/shared_url_loader_factory.h"
+#include "services/network/public/cpp/simple_url_loader.h"
+
+namespace playlist {
+namespace {
+
+const base::FilePath::StringType kBaseDirName(FILE_PATH_LITERAL("playlist"));
+
+constexpr unsigned int kRetriesCountOnNetworkChange = 1;
+
+const base::FilePath::StringType kThumbnailFileName(
+    FILE_PATH_LITERAL("thumbnail"));
+
+void DeleteDir(const base::FilePath& path) {
+  base::DeleteFile(path, true);
+}
+
+PlaylistInfo CreatePlaylistInfo(const CreatePlaylistParams& params) {
+  PlaylistInfo p;
+  p.id = base::Token::CreateRandom().ToString();
+  p.playlist_name = params.playlist_name;
+  p.create_params = params;
+  return p;
+}
+
+base::Value GetValueFromMediaFile(const MediaFileInfo& info) {
```

Comment:
maybe we can move these functions to a separate file, just to keep this one smaller?